File: Lecture6Systems/Env/env.cpp
#include <iostream>
#include <cstdlib>
#include <string>

int main()
{
  std::cout << "Program starting \n";
  auto env = std::getenv("ENV_STARTUP");
  if (env == nullptr)
  {
    std::cout << "You need to set ENV_STARTUP\n";
    exit(EXIT_FAILURE);
  }
  else
  {
    std::cout << "Working dir is " << env << '\n';
  }
  return EXIT_SUCCESS;
}

File: Lecture6Systems/Env/env.py
#!/usr/bin/env python

import os
import sys

if __name__ == "__main__":

    if os.environ.get("ENV_STARTUP") is None:
        print("Environment ENV_STARTUP not set")
        sys.exit(os.EX_CONFIG)
    else:
        print(f" working dir {os.environ['ENV_STARTUP']} ")
